flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -342,21 +343,50 @@ public final OperatorSnapshotResult snapshotState(
 		StateSnapshotContextSynchronousImpl snapshotContext = new StateSnapshotContextSynchronousImpl(
 				checkpointId, timestamp, streamFactory, keyGroupRange, getContainingTask().getCancelables());
 
-		snapshotState(snapshotContext);
+		try {
```

Comment:
Here we could use the try-with-resource if `SnapshotContext` implements `AutoCloseable`